synapse/util/caches/descriptors.py

```diff
@@ -341,7 +370,10 @@ def onErr(f):
                 cache.set(cache_key, result_d, callback=invalidate_callback)
                 observer = result_d.observe()
 
-            return logcontext.make_deferred_yieldable(observer)
+            if isinstance(observer, defer.Deferred):
```

Comment:
`make_deferred_yieldable` will work ok with a non-deferred, so I think this is redundant. otoh I guess it optimises the common path?